include/seqan3/alphabet/all.hpp

```diff
@@ -39,9 +39,127 @@
  *
  * \defgroup alphabet Alphabet
  *
- * The alphabet module contains different biological alphabets and related functionality.
+ * ## Introduction
+ *
+ * Alphabets are a core component in SeqAn. They enable us to represent the smallest
+ * unit of biological sequence data, e.g. a nucleotide or an amino acid.
+ *
+ * In theory, these could just be represented as a `char` and this is how many people
+ * perceive them, but it makes sense to use a smaller, stricter and well-defined alphabet
+ * in almost all cases, because:
+ *
+ *   * Most of our sequences alphabets are actually smaller and can possibly be **represented by less bits**
+ * than a `char`, e.g. a `char` can have 256 values and thus must be represented by 8 bits of memory, but a DNA
+ * character *could* be represented by 2 bits, because it only has four values in the smallest representation
+ * ('A', 'C', 'G', 'T').
+ *   * From a programmer's point of view it is very important to also access the **rank of a letter**, i.e. we need to
+ * be able to convert 'A', 'C', 'G', 'T' to `0`, `1`, `2`, `3` respectively. In fact the rank representation is used a
+ * lot more often than the visual representation which is only used in input/output.
+ *   * You may want to prevent the user from selecting **letters that aren't part of that alphabet**, e.g. 'J' is
+ * not part of any nucleotide definition so people shouldn't be able to select 'J'. With a specialised
+ * alphabet you could instead convert those to a special "unknown"-character, e.g. you might want 'J' to be
+ * always converted to 'N'.
+ *
+ * In SeqAn there are alphabet types for typical sequence alphabets like DNA and amino acid, but also
+ * for qualities, RNA structures and alignment gaps. In addition there are templates for combining alphabet
+ * types into new alphabets, and wrappers for existing data types like the canonical `char`.
+ *
+ * To be included into the alphabet module, an alphabet must satisfy the generic seqan3::alphabet_concept
+ * documented below. While this only encompasses a minimum set of requirements, many of our alphabets provide
+ * more features and there more refined concepts. The inheritance diagram of seqan3::alphabet_concept gives
+ * a detailed overview. A more basic overview of this module and it's submodules is available in the collaboration
+ * diagram at the top of this page.
+ *
+ * ## The alphabet concept
+ *
+ * The seqan3::alphabet_concept defines the requirements a type needs to meet to be considered an alphabet
+ * by SeqAn, or in other words: you can expect certain properties and functions to be defined on
+ * all data types we call an alphabet.
+ *
+ * ### alphabet size
```

Comment:
Rule of thumb for captions and headers is to use capital letters at least for the first word.
There are some more rules, but at least I would request this as a minimum.